ci/cloudbuild/builds/lib/bazel.sh

```diff
@@ -40,3 +40,17 @@ function bazel::common_args() {
   fi
   printf "%s\n" "${args[@]}"
 }
+
+# Bazel downloads all the dependencies of a project, as well as a number of
+# development tools during startup. In automated builds these downloads fail
+# from time to time due to transient network problems. Running `bazel fetch` at
+# the beginning of the build prevents such transient failures from flaking the
+# build.
+function bazel::fetch_dependencies() {
```

Comment:
Do we want this function to be called by all of our bazel builds? Basically by everything that sources this file? I tend to think so. In which case, I'm thinking that we may want to just list this command above (but after the include guard) outside of a function so that it's executed automatically when this file is sourced. We do something similar in the [`lib/cmake.sh`](https://github.com/googleapis/google-cloud-cpp/blob/4755d7ad262d25f430533ae84dccb3764b80badd/ci/cloudbuild/builds/lib/cmake.sh#L30) library. WDYT?